tests/foreman/ui/test_rhcloud_insights.py

```diff
@@ -0,0 +1,98 @@
+"""Tests for RH Cloud - Inventory
+
+:Requirement: RH Cloud - Inventory
+
+:CaseAutomation: Automated
+
+:CaseLevel: System
+
+:CaseComponent: RHCloud-Inventory
+
+:Assignee: jpathan
+
+:TestType: Functional
+
+:CaseImportance: Critical
+
+:Upstream: No
+"""
+from datetime import datetime
+from datetime import timedelta
+
+import pytest
+
+from robottelo.api.utils import wait_for_tasks
+from robottelo.config import settings
+
+
+@pytest.mark.run_in_one_thread
+@pytest.mark.tier3
+def test_rhcloud_insights_e2e(
+    session, rhel8_insights_vm, fixable_rhel8_vm, organization_ak_setup, enable_lab_features
+):
+    """Synchronize hits data from cloud, verify it is displayed in Satellite and run remediation.
+
+    :id: d952e83c-3faf-4299-a048-2eb6ccb8c9c2
+
+    :Steps:
+        1. Prepare misconfigured machine and upload its data to Insights.
+        2. Add Cloud API key in Satellite.
+        3. In Satellite UI, Configure -> Insights -> Add RH Cloud token and syns recommendations.
+        4. Run remediation for dnf.conf recommendation against rhel8 host.
+        5. Assert that job completed successfully.
+        6. Sync Insights recommendations.
+        7. Search for previously remediated issue.
+
+    :expectedresults:
+        1. Insights recommendation related to dnf.conf issue is listed for misconfigured machine.
+        2. Remediation job finished successfully.
+        3. Insights recommendation related to dnf.conf issue is not listed.
+
+    :CaseAutomation: Automated
+    """
+    org, ak = organization_ak_setup
+    query = 'dnf.conf'
+    job_query = (
+        f'Remote action: Insights remediations for selected issues on {rhel8_insights_vm.hostname}'
+    )
+    with session:
+        session.organization.select(org_name=org.name)
+        session.cloudinsights.save_token_sync_hits(settings.rh_cloud.token)
+        # This is a workaround for BZ#1983575. Remove following two code lines once BZ is fixed.
+        session.browser.refresh()
+        session.cloudinsights.sync_hits()
+
+        timestamp = (datetime.utcnow() - timedelta(minutes=2)).strftime('%Y-%m-%d %H:%M')
+        wait_for_tasks(
+            search_query=f'Insights full sync and started_at >= "{timestamp}"',
+            search_rate=15,
+            max_tries=10,
+        )
+        result = session.cloudinsights.search(query)[0]
+        assert result['Hostname'] == rhel8_insights_vm.hostname
+        assert (
+            result['Recommendation'] == 'The dnf installs lower versions of packages when the '
+            '"best" option is not present in the /etc/dnf/dnf.conf'
+        )
+        session.cloudinsights.remediate(query)
+        wait_for_tasks(
```

Comment:
I see the code first tries to check some task started in time t-2 minutes, then it checks whether some other task started in time t. My question now is: Is this reliable pattern? Sorry but I have doubts. 

I would recommend recommend to work on not checking the task being started at specific time.

Reply:
@jaryn timestamp is for searching task started after that time. I had added t-2 minutes by mistake.
It's reliable because along with timestamp, I am using job_query, which is combination of name of task and hostname of the client for which it's been executed.

Reply:
The second task is just for assertion purpose. First one will wait for task to finish, second one is to get/read status of the task i.e if the task finished successfully because first one doesn't provide the right data. Maybe I should find another way to get that info.

Reply:
The test now checks the same thing twice (from line 84 until line 93). That is unnecessary. Please fix

Then on the line 94 it checks the result is success. I don't see any waiting for successful result. Only for job start.
Same goes on line 104. I see there only single refresh is done and then there is an assert for some state. Shouldn't there be some waiting for reaching the state being done?

Reply:
updated the pr.